### ISSUE TITLE
Codesandbox

### ISSUE THREAD

**Issue by jakobrosenberg** [thgh/rollup-plugin-livereload#37]
Hi is it possible to use `livereload` on ``codesandbox`` ?

I'm getting the following error:
```
Failed to execute 'postMessage' on 'DOMWindow': The target origin provided ('<URL>') does not match the recipient window's origin ('<URL>').
```

**Comment by thgh** [comment]
Not sure how codesandbox would support that.

Do you have an example project?

**Comment by jakobrosenberg** [comment]
@thgh, I've created an example here https://codesandbox.io/s/svelte-template-ktznl

It's a 1:1 copy of the official [Svelte template](https://github.com/sveltejs/template).

**Comment by thgh** [comment]
I'm getting `...:35729/livereload.js?snipver=1` `net::ERR_CONNECTION_REFUSED` so it seems the livereload port is not open.

How do other projects handle this?

**Comment by jakobrosenberg** [comment]
I'm not sure, but I think each port is proxied through ``port-hash.sse.codesandbox.io``.

**Comment by thgh** [comment]
Try setting `clientUrl` in the plugin options.

**Comment by jakobrosenberg** [comment]
I tried adding this to ``index.html``
```html
      document.write(
        '<script async src="https://' +
          location.host.replace(/^([^\.]+)/, "$1-35729") +
          '/livereload.js?snipver=1"></' +
          "script>"
      );
```

But then I get these errors when updating files.
``Failed to execute 'postMessage' on 'DOMWindow': The target origin provided ('https://ktznl-35729.sse.codesandbox.io') does not match the recipient window's origin ('https://ktznl.sse.codesandbox.io').``

Setting the ``clientUrl``, would unfortunately require manual configuration for each project which goes against what I'm trying to do.

**Comment by thgh** [comment]
Try adding `?host=ktznl-35729.sse.codesandbox.io` to the livereload script. (found on https://www.npmjs.com/package/livereload-js)

Let's first get it working with manual configuration and then look if the plugin can support it.

**Comment by jakobrosenberg** [comment]
This is the current script
```javascript
      const host = location.host.replace(/^([^\.]+)/, "$1-35729");
      const src = `https://${host}/livereload.js?snipver=1&host=${host}`;
      document.write(`<script async src="${src}">\x3C/script>`);
```

which produces
```html
<script async="" src="https://ktznl-35729.sse.codesandbox.io/livereload.js?snipver=1&host=ktznl-35729.sse.codesandbox.io></script>
```

And this error:
``livereload.js?snipver=1&host=ktznl-35729.sse.codesandbox.io:747 WebSocket connection to 'wss://ktznl-35729.sse.codesandbox.io:35729/livereload' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED``

Is it possible to prevent the rollup plugin from writing a script tag? This would help with filtering the errors.

**Comment by thgh** [comment]
You can copy src/index.js to your project, edit as needed and import that instead. Like this: https://codesandbox.io/s/svelte-template-mo0zo

It looks like you are almost there, it seems to connect to the wrong port:
```diff
      const host = location.host.replace(/^([^\.]+)/, "$1-35729");
-     const src = `https://${host}/livereload.js?snipver=1&host=${host}`;
+     const src = `https://${host}/livereload.js?snipver=1&host=${host}&port=80`;
      document.write(`<script async src="${src}">\x3C/script>`);
```

**Comment by jakobrosenberg** [comment]
```
livereload.js?snipver=1&host=ktznl-35729.sse.codesandbox.io&port=80:747 
WebSocket connection to 'wss://ktznl-35729.sse.codesandbox.io:80/livereload' failed: Error in connection establishment: net::ERR_SSL_PROTOCOL_ERROR
```
Can you make anything from that?

**Comment by jakobrosenberg** [comment]
Changed port to 443 and now it works! :D

**Comment by jakobrosenberg** [comment]
One thing though. Rollup-plugin inserts a script that doesn't work and just produces errors. Is it possible to disable it?

**Comment by thgh** [comment]
Can you make a PR that inserts the correct script based on `process.env.CODESANDBOX_SSE` or maybe `process.env.SANDBOX_URL`?

**Comment by thgh** [comment]
Nice work! Released in v1.1.0 🎉
